### ISSUE TITLE
CORS

### ISSUE THREAD

**Issue by glazec** [tmm/wagmi#174]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Package Version

0.2.9

### Current Behavior

Access to fetch at 'https://cloudflare-eth.com/' from origin 'https://xxxx.vercel.app' has been blocked by CORS policy: Response to preflight request doesn't pass access control check: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.

POST https://cloudflare-eth.com/ net::ERR_FAILED

### Expected Behavior

Successfully read contract

### Steps To Reproduce

```
const NFTId = () => {
  const provider = useProvider()
  const [{ data, error, loading }, read] = useContractRead(
    {
      addressOrName: '0xe538B351070C02d17b5878D3017932A68606fbaE',
      contractInterface: passABI,
      signerOrProvider: provider,
    },
    'totalSupply',
    {
      args: [0],
    }
  )
  console.log(data)
  return (
    <div>
      <span className="nes-text">{loading
        ?
        null
        : data}</span>
    </div>
  )
}
```

**Comment by tmm** [comment]
Hi! Can you create a CodeSandbox ([template](https://codesandbox.io/s/wagmi-template-d2ct0?file=/src/App.tsx)) with this or post the ABI?

**Comment by glazec** [comment]
I think I found the issue. The problem is that the web app failed to call the view function of the contract and the web app is keep retrying this. I found that useContractRead is set to default to read mainnet contract. 

If I would like to read a contract from rinkeby, how should I configure it?

**Comment by tmm** [comment]
Set up the ethers.js provider so it is [chain-aware](https://wagmi-xyz.vercel.app/docs/provider#provider-optional).

**Comment by ykchong45** [comment]
I faced similar issue, I used `useBlocknumber()` to check if my app is connected to a correct chain. 

For me I was running on mumbai testnet, it returns an `undefined` for block number. The solution was to follow [configure chains](https://wagmi.sh/docs/providers/configuring-chains)' instruction, and add in chains that I wanted to connect, i.e. `[chain.mainnet, chain.polygonMumbai]`, instead of feeding in custom RPC (error-prone).
